File: FactorAnalysis/ClientApp/src/app/main/forecasting-task/models/analyze-prediction-algorithms-report.ts
import { LearningAlgorithm } from './learning-algorithm.enum';
import { AlgorithmPredictionResult } from './algorithm-prediction-result';

export class AnalyzePredictionAlgorithmsReport {
    algorithm: LearningAlgorithm;
    elapsedTrainingTime: Date;
    elapsedPredictionTime: Date;
    averageError: number;
    results: AlgorithmPredictionResult[];
}

File: FactorAnalysis/Model/AnalyzePredictionAlgorithmsReport.cs
using DomainModel.ForecastingTasks;
using System;
using System.Collections.Generic;

namespace FactorAnalysis.Model
{
    public class AnalyzePredictionAlgorithmsReport
    {
        public LearningAlgorithm Algorithm { get; set; }

        public TimeSpan ElapsedTrainingTime { get; set; }

        public TimeSpan ElapsedPredictionTime { get; set; }

        public double AverageError { get; set; }

        public List<AlgorithmPredictionResult> Results { get; set; }
    }
}
